### PR TITLE
persistent-cache-cpp: init at 1.0.5

### PR DESCRIPTION
**Pull request by OPNA2608**
## Description of changes

Working towards https://github.com/NixOS/nixpkgs/issues/99090.

https://gitlab.com/ubports/development/core/lib-cpp/persistent-cache-cpp, used by Lomiri's Thumbnailer service.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/pe/persistent-cache-cpp/package.nix [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    # Remove when version > 1.0.5
+    (fetchpatch {
+      url = "https://gitlab.com/ubports/development/core/lib-cpp/persistent-cache-cpp/-/commit/a590ffcccec252caa7b19a2922c678502069b057.patch";
+      hash = "sha256-OhlsnUVhclt17brkncYkJ0+lXrJO671mtwopaAGPrtY=";
+    })
+
+    # PersistentStringCacheImpl.exceptions test fails on LLVM's libcxx, it depends on std::system_error producing a very specific exception text
+    # Expects "Unknown error 666", gets "unspecified generic_category error"
+    # https://gitlab.com/ubports/development/core/lib-cpp/persistent-cache-cpp/-/blob/1.0.5/tests/core/internal/persistent_string_cache_impl/persistent_string_cache_impl_test.cpp?ref_type=tags#L1298
+    ./0001-persistent-cache-cpp-Lenient-exception-test-matching.patch
```
Comment (pbsds) [inline pkgs/by-name/pe/persistent-cache-cpp/package.nix:46]: This looks very upstreamable. Would you be willing to?

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 259393` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>persistent-cache-cpp</li>
  </ul>
</details>

The build seems to produce a `.a` file for static compilation, and not a dynamic `.so`. Is this expected?

**Comment by OPNA2608** [comment]
Slightly outdated, [1.0.6 was released](https://gitlab.com/ubports/development/core/lib-cpp/persistent-cache-cpp/-/tags/1.0.6) since this was opened. I'll work on a bump.

> The build seems to produce a `.a` file for static compilation, and not a dynamic `.so`. Is this expected?

Expected insofar that upstream [explicitly builds a static library](https://gitlab.com/ubports/development/core/lib-cpp/persistent-cache-cpp/-/blob/bd3a68d72fc890a88afffecfe4647988bf23dce8/src/core/CMakeLists.txt#L11), and I haven't checked if forcing it to be whatever the stdenv prefers breaks anything further down the line. I can give it a try though.

File: pkgs/by-name/pe/persistent-cache-cpp/package.nix (hunk 1 above)
Comment (OPNA2608) [reply pkgs/by-name/pe/persistent-cache-cpp/package.nix:46]: Can do.

**Comment by OPNA2608** [comment]
Sorry, took me sometime to get around to this. Bumped to new version, and submitted afew changes upstream which we now `fetchpatch`.

(The `libdir` patching in the pkg-config file should also be upstreamable, but I didn't remember the correct way of approaching that OTTOMH)

**Comment by OPNA2608** [comment]
> `error: getting status of '/private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/macstadium-m1-44911104/pkgs/by-name/pe/persistent-cache-cpp': No such file or directory`

Not sure what to make of this aarch64-darwin CI failure. x86_64-darwin builds fine FWIW.

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 259393` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>persistent-cache-cpp</li>
    <li>persistent-cache-cpp.dev</li>
    <li>persistent-cache-cpp.doc</li>
  </ul>
</details>


- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [ ] executables tested on x86_64-linux
  - just a lib, but `env -i ldd` is happy
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

**Pull request merged**